gramps/gen/filters/rules/person/_hascommonancestorwith.py

```diff
@@ -96,17 +106,22 @@ def add_ancs(self, db, person):
     def reset(self):
         self.ancestor_cache = {}
 
-    def has_common_ancestor(self, other):
+    def has_common_ancestor(self, other: Person):
         for handle in self.with_people:
-            if (handle in self.ancestor_cache and self.ancestor_cache[handle]) & (
+            # Requires bitwise and for some reason:
+            left_and = (
+                handle in self.ancestor_cache and self.ancestor_cache[handle]
+            )  # type: ignore
+            right_and = (
                 other
                 and other.handle in self.ancestor_cache
                 and self.ancestor_cache[other.handle]
-            ):
+            )  # type: ignore
+            if left_and & right_and:  # type: ignore
```

Comment:
```suggestion
            if left_and.intersection(right_and):  # type: ignore
```
use `intersection()` in preference to `&`